From: Alessandro Zummo <alessandro.zummo@towertech.it>
Date: Sun, 18 Mar 2007 14:23:33 +0000 (+0100)
Subject: pata_ixp4xx_cf: fix interrupt
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=282c6b9c7a344ce6436bb61a006b9c2a373b9626

pata_ixp4xx_cf: fix interrupt

 As per compact flash specifications, the default
 irq mode upon cf insertion is pulse mode. this patch fixes
 the driver to cope with that.

Signed-off-by: Alessandro Zummo <a.zummo@towertech.it>
Signed-off-by: Jeff Garzik <jeff@garzik.org>
---

Index: linux-2.6.20-arm/drivers/ata/pata_ixp4xx_cf.c
===================================================================
--- linux-2.6.20-arm.orig/drivers/ata/pata_ixp4xx_cf.c	2007-02-04 18:44:54.000000000 +0000
+++ linux-2.6.20-arm/drivers/ata/pata_ixp4xx_cf.c	2007-03-22 09:16:09.653361584 +0000
@@ -200,7 +200,7 @@
 
 	irq = platform_get_irq(pdev, 0);
 	if (irq)
-		set_irq_type(irq, IRQT_HIGH);
+		set_irq_type(irq, IRQT_RISING);
 
 	/* Setup expansion bus chip selects */
 	*data->cs0_cfg = data->cs0_bits;
